spec/features/welcome_mailer_spec.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+require 'spec_helper'
+
+describe WelcomeMailer do
+  let(:user_email) { 'example@test.com' }
+  let(:user_password) { 'really_good_password' }
+  let(:email) { ActionMailer::Base.deliveries.last }
+  before do
+    visit new_user_registration_path
+
+    fill_in 'user_email', with: user_email
+    fill_in 'user_password', with: user_password
+    fill_in 'user_password_confirmation', with: user_password
+
+    click_button 'Sign up'
+  end
+  after do
+    ActionMailer::Base.deliveries = []
+    User.find_by_email(user_email).delete
+  end
+  it 'sends welcome email to user upon registration' do
+    email.to.should eq([user_email])
+    email.from.should == ["scholar@uc.edu"]
+  end
+  describe 'Shibboleth Logins' do
```

Comment:
@13miciunasl Instead of starting a new "Shibboleth Logins" describe block, break the code below into two separate context blocks for students and non-students.